File: ra-back/src/main/java/pl/raptors/ra_back/domain/mapsMetadata/Maps.java
package pl.raptors.ra_back.domain.mapsMetadata;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@NoArgsConstructor
@Data
@Document(collection = "map_metadata")
public class Maps {
    @Id
    String id;
    String urlYaml;
    String urlPng;
    String urlPgm;
    String[]tags;

    MapMetadata mapMetadata;
    MapTransformation mapTransformation;
    MapRotation mapRotation;

    public Maps(String urlYaml, String urlPng, String urlPgm, String[] tags, MapMetadata mapMetadata, MapTransformation mapTransformation, MapRotation mapRotation) {
        this.urlYaml = urlYaml;
        this.urlPng = urlPng;
        this.urlPgm = urlPgm;
        this.tags = tags;
        this.mapMetadata = mapMetadata;
        this.mapTransformation = mapTransformation;
        this.mapRotation = mapRotation;
    }
}

File: ra-front/src/app/model/mapsMetadata/Maps.ts
import {MapMetadata} from "./MapMetadata";
import {MapTransformation} from "./MapTransformation";
import {MapRotation} from "./MapRotation";

export class Maps {
        id: string;
        urlYaml: string;
        urlPng: string;
        urlPgm: string;
        tags: string[];

         mapMetadata: MapMetadata;
         mapTransformation: MapTransformation;
         mapRotation: MapRotation;


  constructor() {
    this.id = "";
    this.urlYaml = "";
    this.urlPng = "";
    this.urlPgm = "";
    this.tags = [];
    this.mapMetadata = new MapMetadata("", "", "", [], "", "", "");
    this.mapTransformation = new MapTransformation("", 0, 0, 0);
    this.mapRotation = new MapRotation("", 0);
  }
}
